robocop/checkers/naming.py

```diff
@@ -350,6 +362,36 @@
         is available at https://github.com/robotframework/robotframework/issues/4431 .
         """,
     ),
+    "0323": Rule(
+        rule_id="0323",
+        name="inconsistent-variable-name",
+        msg="Variable '{{ name }}' has inconsistent naming. First used as '{{ first_use }}'",
+        severity=RuleSeverity.WARNING,
+        docs="""
+        Variable names are case-insensitive and ignore underscores and spaces. It is possible to 
+        write the variable in multiple ways and it will be a valid Robot Framework code. However
+        it makes it harder to maintain the code that is not following consistent naming.
```

Comment:
```suggestion
        Variable names are case-insensitive and ignore underscores and spaces. It is possible to 
        write the variable in multiple ways and it will be a valid Robot Framework code. However,
        it makes it harder to maintain the code that does not follow the consistent naming.
```